### PR TITLE
New module: xmlrpc_client

### PR DESCRIPTION
**Pull request by stemid**
##### SUMMARY

I have an API internally where this module helps a lot because it lets me use ansible for the entire deployment process. 

It's the most basic xmlrpc API client that forwards any call to the API along with positional arguments and keyword arguments. This way the ansible user can decide how to use the xmlrpc api along with the relevant API docs.

Looking at the webfaction modules for example I can see that a basic xmlrpc client module would be able to perform their tasks as well but with less abstractions between the user and the API.

I wasn't sure which category to put the module in. I'm sure people will have issue with placing it under commands, in that case just know that I have no problem with moving it to another category if a more suitable one can be found.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

lib/ansible/modules/commands/xmlrpc_client
##### ANSIBLE VERSION

```
ansible 2.4.0 (xmlrpc_client-module 5224579d9e) last updated 2017/08/11 16:08:55 (GMT +200)
  config file = None
  configured module search path = ['/home/stemid/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/stemid/Utveckling/ansible/lib/ansible
  executable location = /home/stemid/Utveckling/ansible/bin/ansible
  python version = 3.6.2 (default, Jul 19 2017, 13:09:21) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]

```

##### ADDITIONAL INFORMATION

I have run sanity checks and there are no more issues with my code but some sanity checks in ansible-test still fail because of other older modules in the repo. So I expect ansibot to start posting those here soon. What can I do when all my own sanity checks are resolved?

```

```

### CODE REVIEW

File: lib/ansible/modules/commands/xmlrpc_client.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            - The API URL you connect to.
+        required: true
+    path:
+        description:
+            - The RPC path/method you are calling.
+        required: true
+    args:
+        description:
+            - A list of arguments for the RPC call. Placed first in any RPC call.
+        required: false
```
Comment (dagwieers) [inline lib/ansible/modules/commands/xmlrpc_client.py]: This is not needed for parameters that are not required. Guideline dictates to leave it out.

File: lib/ansible/modules/commands/xmlrpc_client.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    kwargs:
+        description:
+            - A dictionary of arguments for the RPC call. Comes after the args list.
+        required: false
```
Comment (dagwieers) [inline lib/ansible/modules/commands/xmlrpc_client.py]: This is not needed for parameters that are not required. Guideline dictates to leave it out.

File: lib/ansible/modules/commands/xmlrpc_client.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    from xmlrpclib import ServerProxy
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_module():
+    module_args = dict(
+        url=dict(type='str', required=True),
+        path=dict(type='str', required=True),
+        args=dict(type='list', required=False, default=[]),
```
Comment (dagwieers) [inline lib/ansible/modules/commands/xmlrpc_client.py]: Here also `required=False` is not needed for parameters that are not required. Guideline dictates to leave it out.

File: lib/ansible/modules/commands/xmlrpc_client.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        kwargs=dict(type='dict', required=False, default={})
```
Comment (dagwieers) [inline lib/ansible/modules/commands/xmlrpc_client.py]: Here also `required=False` is not needed for parameters that are not required. Guideline dictates to leave it out.

**Review by dagwieers — CHANGES_REQUESTED** [review]
Looking good.

**Comment by stemid** [comment]
@dagwieers Requested changes made, committed and sanity checks still report ok for this module.

**Comment by stemid** [comment]
I made a mistake with the rebase and now we have 20 unnecessary participants so I'm closing this and opening a new one. Sorry for the confusion.

**Comment by dagwieers** [comment]
@stemid You can always fix that by force-pushing the (corrected) branch. The pull-request will be adapted. (Often you can correct your branch using `git rebase -i HEAD~40` removing all commits unrelated to your change up to the point the branch was still in sync with devel.)

**Pull request closed without merging**